File: programmers/H-Index.py
def solution(citations):
    arr = sorted(citations)

    l = len(arr)
    h = max(arr) if max(arr) <= l else l
    if h != 0:
        while h >= 0:
            if arr[l - h] >= h:
                break
            h -= 1

    answer = h
    return answer

File: programmers/H-index.js
// N = 10^3, M = 10^5 -> NM = 10^8, 완탐x

// 인용 횟수(citations) 정렬
// h = n - i
// h번 이상 인용된 논문이 h편 이상, 나머지
// citatinos[i]번 이상 인용된 논문이 (n - i)편 이상

function solution(citations) {
    citations.sort((a, b) => a - b)
    
    const n = citations.length
    let ans = 0
    for (let i=0; i<n; i++) {
        if (citations[i] >= n - i) {
            ans = n - i
            break
        }
    }
    
    return ans;
}